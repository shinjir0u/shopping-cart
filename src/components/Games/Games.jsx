import { useNavigate, useOutletContext } from "react-router-dom";
import style from "./games.module.css";

function Games() {
  const { games } = useOutletContext();
  const navigate = useNavigate();
  const handleClick = (id) => navigate(`game/${id}`);

  return (
    <div className={style.games}>
      <div className={`${style.games__content} container`}>
        {games.map((game) => (
          <Game key={game.id} onClick={() => handleClick(game.id)} game={game} />
        ))}
      </div>
    </div>
  );
}

function Game({ game, onClick }) {
  return (
    <div className={style.game__card} onClick={onClick}>
      <img className={style.game__image} src={game.background_image} alt={game.name} />
      <div className={style.game__information}>
        <h2 className="game__name">{game.name}</h2>
        <p className="game__store">
          {game.stores.map(
            (store, index) =>
              store.store.name + (index === game.stores.length - 1 ? "" : ", ")
          )}
        </p>
      </div>
    </div>
  );
}

export default Games;
