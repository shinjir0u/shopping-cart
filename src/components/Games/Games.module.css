.games__content {
  display: grid;
  gap: 2em;
}

.game__card {
  background-color: var(--brown-color);
  padding: 1em;
  color: var(--white-color);
  box-shadow: 2px 2px 2px var(--dark-background-color);
  border-radius: 10px;
  cursor: pointer;
  display: grid;
  grid-template-rows: minmax(100px, 150px) 1fr;
  gap: 1em;
  transition: transform 0.2s ease-out;
}

.game__card:hover {
  transform: scale(1.05) ;
}

.game__image {
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 10px;
}
