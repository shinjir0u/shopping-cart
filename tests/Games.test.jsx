import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import sampleGames from "../helperFiles/sampleGamesObject";
import Games from "../src/components/Games/Games";

describe("Games component exists in DOM", () => {
  it("Games component renders with the correct data", () => {
    const result = render(<Games games={sampleGames.results} />);
    expect(result).toMatchSnapshot();
  });
});
