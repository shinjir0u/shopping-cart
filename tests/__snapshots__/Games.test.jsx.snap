// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Games component exists in DOM > Games component renders with the correct data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="games"
      >
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/20a/20aa03a10cda45239fe22d035c0ebe64.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Grand Theft Auto V
            </h2>
            <p
              class="game__store"
            >
              Steam, 
              PlayStation Store, 
              Epic Games, 
              Xbox 360 Store, 
              Xbox Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              The Witcher 3: Wild Hunt
            </h2>
            <p
              class="game__store"
            >
              GOG, 
              PlayStation Store, 
              Steam, 
              Xbox Store, 
              Nintendo Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/2ba/2bac0e87cf45e5b508f227d281c9252a.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Portal 2
            </h2>
            <p
              class="game__store"
            >
              Xbox Store, 
              Steam, 
              PlayStation Store, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/736/73619bd336c894d6941d926bfd563946.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Counter-Strike: Global Offensive
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              Steam, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Tomb Raider (2013)
            </h2>
            <p
              class="game__store"
            >
              Xbox 360 Store, 
              Steam, 
              PlayStation Store, 
              Google Play, 
              App Store, 
              Epic Games
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/7fa/7fa0b586293c5861ee32490e953a4996.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Portal
            </h2>
            <p
              class="game__store"
            >
              Steam, 
              Google Play
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/d58/d588947d4286e7b5e0e12e1bea7d9844.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Left 4 Dead 2
            </h2>
            <p
              class="game__store"
            >
              Steam, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/7cf/7cfc9220b401b7a300e409e539c9afd5.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              The Elder Scrolls V: Skyrim
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              Steam, 
              Nintendo Store, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/511/5118aff5091cb3efec399c808f8c598f.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Red Dead Redemption 2
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              Epic Games, 
              Steam, 
              Xbox Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/fc1/fc1307a2774506b5bd65d7e8424664a7.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              BioShock Infinite
            </h2>
            <p
              class="game__store"
            >
              Epic Games, 
              App Store, 
              Xbox Store, 
              Nintendo Store, 
              PlayStation Store, 
              Steam, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/b8c/b8c243eaa0fbac8115e0cdccac3f91dc.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Half-Life 2
            </h2>
            <p
              class="game__store"
            >
              Steam, 
              Google Play
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/49c/49c3dfa4ce2f6f140cc4825868e858cb.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Borderlands 2
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              Google Play, 
              Steam, 
              Xbox 360 Store, 
              App Store, 
              Xbox Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/562/562553814dd54e001a541e4ee83a591c.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Life is Strange
            </h2>
            <p
              class="game__store"
            >
              GOG, 
              PlayStation Store, 
              Steam, 
              Xbox Store, 
              Google Play, 
              Xbox 360 Store, 
              App Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/bc0/bc06a29ceac58652b684deefe7d56099.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              BioShock
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              App Store, 
              Xbox Store, 
              Steam, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Destiny 2
            </h2>
            <p
              class="game__store"
            >
              Epic Games, 
              PlayStation Store, 
              Xbox Store, 
              Steam
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/4be/4be6a6ad0364751a96229c56bf69be59.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              God of War (2018)
            </h2>
            <p
              class="game__store"
            >
              GOG, 
              PlayStation Store, 
              Steam, 
              Epic Games
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/d82/d82990b9c67ba0d2d09d4e6fa88885a7.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Fallout 4
            </h2>
            <p
              class="game__store"
            >
              PlayStation Store, 
              Xbox Store, 
              Steam
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/73e/73eecb8909e0c39fb246f457b5d6cbbe.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              PAYDAY 2
            </h2>
            <p
              class="game__store"
            >
              Epic Games, 
              PlayStation Store, 
              Steam, 
              Nintendo Store, 
              Xbox 360 Store
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/942/9424d6bb763dc38d9378b488603c87fa.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Limbo
            </h2>
            <p
              class="game__store"
            >
              App Store, 
              PlayStation Store, 
              Xbox Store, 
              Steam, 
              GOG, 
              Xbox 360 Store, 
              Nintendo Store, 
              Google Play, 
              Epic Games
            </p>
          </div>
        </div>
        <div
          class="game__card"
        >
          <img
            alt="game image"
            src="https://media.rawg.io/media/games/46d/46d98e6910fbc0706e2948a7cc9b10c5.jpg"
          />
          <div
            class="game__information"
          >
            <h2
              class="game__name"
            >
              Team Fortress 2
            </h2>
            <p
              class="game__store"
            >
              Steam
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="games"
    >
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/20a/20aa03a10cda45239fe22d035c0ebe64.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Grand Theft Auto V
          </h2>
          <p
            class="game__store"
          >
            Steam, 
            PlayStation Store, 
            Epic Games, 
            Xbox 360 Store, 
            Xbox Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            The Witcher 3: Wild Hunt
          </h2>
          <p
            class="game__store"
          >
            GOG, 
            PlayStation Store, 
            Steam, 
            Xbox Store, 
            Nintendo Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/2ba/2bac0e87cf45e5b508f227d281c9252a.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Portal 2
          </h2>
          <p
            class="game__store"
          >
            Xbox Store, 
            Steam, 
            PlayStation Store, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/736/73619bd336c894d6941d926bfd563946.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Counter-Strike: Global Offensive
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            Steam, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Tomb Raider (2013)
          </h2>
          <p
            class="game__store"
          >
            Xbox 360 Store, 
            Steam, 
            PlayStation Store, 
            Google Play, 
            App Store, 
            Epic Games
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/7fa/7fa0b586293c5861ee32490e953a4996.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Portal
          </h2>
          <p
            class="game__store"
          >
            Steam, 
            Google Play
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/d58/d588947d4286e7b5e0e12e1bea7d9844.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Left 4 Dead 2
          </h2>
          <p
            class="game__store"
          >
            Steam, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/7cf/7cfc9220b401b7a300e409e539c9afd5.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            The Elder Scrolls V: Skyrim
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            Steam, 
            Nintendo Store, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/511/5118aff5091cb3efec399c808f8c598f.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Red Dead Redemption 2
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            Epic Games, 
            Steam, 
            Xbox Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/fc1/fc1307a2774506b5bd65d7e8424664a7.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            BioShock Infinite
          </h2>
          <p
            class="game__store"
          >
            Epic Games, 
            App Store, 
            Xbox Store, 
            Nintendo Store, 
            PlayStation Store, 
            Steam, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/b8c/b8c243eaa0fbac8115e0cdccac3f91dc.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Half-Life 2
          </h2>
          <p
            class="game__store"
          >
            Steam, 
            Google Play
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/49c/49c3dfa4ce2f6f140cc4825868e858cb.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Borderlands 2
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            Google Play, 
            Steam, 
            Xbox 360 Store, 
            App Store, 
            Xbox Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/562/562553814dd54e001a541e4ee83a591c.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Life is Strange
          </h2>
          <p
            class="game__store"
          >
            GOG, 
            PlayStation Store, 
            Steam, 
            Xbox Store, 
            Google Play, 
            Xbox 360 Store, 
            App Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/bc0/bc06a29ceac58652b684deefe7d56099.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            BioShock
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            App Store, 
            Xbox Store, 
            Steam, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Destiny 2
          </h2>
          <p
            class="game__store"
          >
            Epic Games, 
            PlayStation Store, 
            Xbox Store, 
            Steam
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/4be/4be6a6ad0364751a96229c56bf69be59.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            God of War (2018)
          </h2>
          <p
            class="game__store"
          >
            GOG, 
            PlayStation Store, 
            Steam, 
            Epic Games
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/d82/d82990b9c67ba0d2d09d4e6fa88885a7.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Fallout 4
          </h2>
          <p
            class="game__store"
          >
            PlayStation Store, 
            Xbox Store, 
            Steam
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/73e/73eecb8909e0c39fb246f457b5d6cbbe.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            PAYDAY 2
          </h2>
          <p
            class="game__store"
          >
            Epic Games, 
            PlayStation Store, 
            Steam, 
            Nintendo Store, 
            Xbox 360 Store
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/942/9424d6bb763dc38d9378b488603c87fa.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Limbo
          </h2>
          <p
            class="game__store"
          >
            App Store, 
            PlayStation Store, 
            Xbox Store, 
            Steam, 
            GOG, 
            Xbox 360 Store, 
            Nintendo Store, 
            Google Play, 
            Epic Games
          </p>
        </div>
      </div>
      <div
        class="game__card"
      >
        <img
          alt="game image"
          src="https://media.rawg.io/media/games/46d/46d98e6910fbc0706e2948a7cc9b10c5.jpg"
        />
        <div
          class="game__information"
        >
          <h2
            class="game__name"
          >
            Team Fortress 2
          </h2>
          <p
            class="game__store"
          >
            Steam
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
